chartpress.py

```diff
@@ -1001,10 +1069,16 @@ def main(args=None):
     #   - push chart (--publish-chart, --extra-message)
     for chart in config["charts"]:
         forced_version = None
+        # TODO: maybe warn and switch default in the future?
+        use_chart_version = chart.get("useChartVersion", False)
+
         if args.tag:
+            # tag specified, use that version
             forced_version = args.tag
-        elif args.reset:
-            forced_version = chart.get("resetVersion", "0.0.1-set.by.chartpress")
+        elif args.reset and not use_chart_version:
+            # resetting, get version from chrtpress.yaml,
+            # ignoring current version in Chart.yaml
+            forced_version = chart.get("resetVersion", "0.0.1-0.dev")
```

Comment:
I don't think this is a breaking change at all. It does change what the version is reset _to_, but it's just as consistent. As you say, the version is never used, and after the chartpress version is upgraded once, it continues to be idemponent.

I can change it back, if you think `set.by.chartpress` is better.

Reply:
Okay hmmm, yeah I think `set.by.chartpress` is better and I'm not sure what to be considered a breaking change - but I know this change would make the z2jh that relies on the default value need a change to help its CI system running `chartpress --reset` not report failures where `0.0.0-set.by.chartpress` as version controlled would change to `0.0.1-0.dev` instead.

I care mostly because I think the placeholder values in z2jh is more comprehensible like this than if they are set to `0.0.1-dev.0`, and, when this setting is disabled, they will be placeholders rather than a value to be used.

Reply:
Fair enough, I'll put it back.